modules/tensor_mechanics/src/kernels/GeneralizedPlaneStrainOffDiag.C

```diff
@@ -63,9 +63,7 @@ GeneralizedPlaneStrainOffDiag::GeneralizedPlaneStrainOffDiag(const InputParamete
     _scalar_var_id(isParamValid("scalar_out_of_plane_strain_index")
                        ? getParam<unsigned int>("scalar_out_of_plane_strain_index")
                        : 0),
-    _temp_var(isParamValid("temperature")
-                  ? &_subproblem.getStandardVariable(_tid, getParam<VariableName>("temperature"))
-                  : NULL),
+    _temp_var(isCoupled("temperature") ? getVar("temperature", 0) : nullptr),
```

Comment:
```suggestion
    _temp_var(isCoupled("temperature") ? coupled("temperature") : libMesh::invalid_uint),
```

and make `_temp_var` an `unsigned int`

Reply:
We need the variable names, though, for the call to `getMaterialPropertyDerivatives()`. How do you get the name from the variable index?  Storing the variable pointer doesn't seem like a problem to me, and it seems consistent with how this is done in other classes that use material property derivatives.

Reply:
I've never seen it done that way, but if it works it's fine with me.